target/linux/ramips/dts/mt7620a_edimax_ew-7476hpc-16m.dts

```diff
@@ -0,0 +1,249 @@
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/dts-v1/;
+
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/mtd/partitions/uimage.h>
+
+/ {
+	compatible = "edimax,ew-7476hpc-16m", "ralink,mt7620a-soc";
+	model = "Edimax EW-7476HPC (CAP1200 16M)";
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		wlan2g {
+			label = "cap1200:blue:wlan2g";
+			gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan5g {
+			label = "cap1200:yellow:wlan5g";
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	beeper: beeper {
+		compatible = "gpio-beeper";
+		gpios = <&gpio2 12 GPIO_ACTIVE_LOW>;
+	};
+	
+	virtual_flash {
+		compatible = "mtd-concat";
+
+		devices = <&fwconcat0 &fwconcat1 &fwconcat2>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "firmware";
+				reg = <0x0 0x0>;
+				compatible = "denx,uimage";
+			};
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partition@0 {
+			label = "u-boot";
+			reg = <0x0 0x30000>;
+			read-only;
+		};
+
+		config: partition@30000 {
+			label = "config";
+			reg = <0x30000 0x10000>;
+			read-only;
+		};
+		
+		factory: partition@40000 {
+			label = "factory";
+			reg = <0x40000 0x8000>;
+			read-only;
+		};
+
+		art: partition@48000 {
+			label = "art";
+			reg = <0x48000 0x8000>;
+			read-only;
+		};
+
+		fwconcat0:partition@50000 {
+			label = "fwconcat0";
+			reg = <0x50000 0x179000>;
+		};
+		
+		fwconcat1:partition@1c9000 {
+			label = "fwconcat1";
+			reg = <0x1c9000 0xB11000>;
+		};
+		
+		partition@cda000 {
+			label = "nms_data";
+			reg = <0xcda000 0x200000>;
+		};
+
+		partition@eda000 {
+			label = "manufacture";
+			reg = <0xeda000 0x96000>;
+		};
+
+		partition@f70000 {
+			label = "backup";
+			reg = <0xf70000 0x10000>;
+		};
+
+		fwconcat2:partition@7c0000 {
+			label = "fwconcat2";
+			reg = <0xf80000 0x80000>;
+		};
+	};
+};
+
+&state_default {
+	gpio {
+		groups = "i2c", "uartf", "nd_sd", "rgmii2";
+		function = "gpio";
+	};
+};
+
+&pinctrl {
+	phy_reset_pins: phy-reset {
+		gpio {
+			groups = "spi refclk";
+			function = "gpio";
+		};
+	};
+};
+
+&ethernet {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rgmii1_pins &mdio_pins &phy_reset_pins>;
+
+	mtd-mac-address = <&factory 0x4>;
```

Comment:
Nothing in the other locations? 0x28, 0x2e, 0xe000, 0xe006 ...

Reply:
Will retest

Reply:
root@OpenWrt:/# hexdump -C /dev/mtd2 | grep "74 da 38"
00000000  20 76 08 01 74 da 38 5e  a1 62 ff ff ff ff ff ff  | v..t.8^.b......|

There is only mac related information at 0x04